Add ed.json for instant answers on ed(1)

Gumnos opened the pull request:
IA Page: https://duck.co/ia/view/ed_cheat_sheet

moollaza commented:
@Gumnos thanks a lot for contributing, we really appreciate it!

Can you please fill out the [pull request template](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/pull_request_template_spice.md) and edit your initial comment. Once completed, we'll be sure to review your PR and give you some feedback.

Thanks!

Gumnos commented:
# Spice Pull Request Template

**What does your Instant Answer do?**

Provides a cheat-sheet for `ed(1)`

**What problem does your Instant Answer solve (Why is it better than organic links)?**

Has `ed(1)` commands as well as details differentiating GNU `ed(1)` from BSD `ed(1)`

**What is the data source for your Instant Answer? (Provide a link if possible)**

http://www.gnu.org/software/ed/manual/ed_manual.html
https://www.freebsd.org/cgi/man.cgi?ed%281%29

**Why did you choose this data source?**

Two authoritative sources on GNU & BSD `ed(1)`

**Are there any other alternative (better) data sources?**

Not that I've found

**What are some example queries that trigger this Instant Answer?**

"ed cheatsheet" or any of the other cheat-sheet triggers

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Unix/Linux/BSD geeks

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

Not as far as I know.

**Which existing Instant Answers will this one supersede/overlap with?**

Augments the "cheat sheets"

**Are you having any problems? Do you need our help with anything?**

After some experimentation on Codio, it came out how I wanted.

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**

N/A

**Where did you hear about DuckDuckHack? (For first time contributors)**

Twitter @duckduckhack account

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

A cheat-sheet.

## Checklist
Please place an 'X' where appropriate.

Since it only adds a new cheat-sheet to the existing functionality, the remainder should already have been tested.  I did test against the Codio server in Chromium & Firefox on Linux and they came out fine.

```
[x] Added metadata and attribution information
[] Wrote test file and added to t/ directory
[] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
```

Also, I didn't see any contribution/license arrangement/agreement terms, but consider my diff/patch under whatever common permissive license you need/want. BSD, WTFPL, CC0, etc.

moollaza commented on share/goodie/cheat_sheets/ed.json:
```diff
@@ -0,0 +1,238 @@
+    "sections": {
+        "Invocation": [
+            {
+                "key": "[ed\u00a0\\[parameters\\]\u00a0\\[filename\\]]",
```
Please remove the `\u00a0` -- it's expected to wrap. We're working towards making sure it always still looks good, so it's alright for now if it doesn't look perfect :)

moollaza commented on share/goodie/cheat_sheets/ed.json:
```diff
@@ -0,0 +1,238 @@
+                "val": "A line marked with {letter} (see the [k] command below)"
+            }
+        ],
+        "Commands": [
```
Any chance you can break this down into more than 1 section? Do these commands fall into any kind of sub-categories? 

Having more sections makes it easier to layout the content across the width of the screen, and looks better too :smile:

moollaza commented:
@Gumnos this is looking good. Just a few more things to address and this one should be good :+1:

Gumnos commented:
@moollaza  How would you like the resulting changes? As a sequence of git commits atop this pull-request, as one single squashed commit atop this pull-request, or as a fresh branch based off `origin/master`?

mintsoft commented:
@Gumnos commits atop the PR is usual; however you can squash it if you'd prefer

Gumnos commented:
I've added the commits atop this one.  They were two commits, one for each requested change, but squashed for simplicity. I can split them back out if you more atomic changes.

moollaza commented:
@Gumnos Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

moollaza commented:
@Gumnos congrats! It's live: https://duckduckgo.com/?q=ed+cheat+sheet&ia=cheatsheet

Sorry for the late notice, it's been live for a few days I believe :)

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!
